src/providers/connectedClusters.ts

```diff
@@ -0,0 +1,159 @@
+	private kubeConfig = new k8s.KubeConfig();
+	private connectedClusters: ConnectedCluster[] = [];
+	private configRefreshedEmitter: vscode.EventEmitter<void> = new vscode.EventEmitter<void>();
+	readonly onDidChangeTreeData: vscode.Event<void> = this.configRefreshedEmitter.event;
+
+	constructor() {
+		this.updateK8SConfig();
+	}
+
+	refresh(): void {
+		this.updateK8SConfig();
+		this.configRefreshedEmitter.fire();
+	}
+
+	getTreeItem(element: CustomViewType): vscode.TreeItem | Thenable<vscode.TreeItem> {
+		return element;
+	}
+
+	getChildren(element?: CustomViewType): vscode.ProviderResult<CustomViewType[]> {
+		if (element) {
+			if (element instanceof ConnectedCluster) {
+				// children of a connected clusters are ocm's api resources
+				return this.getOcmResourceDefinitions(element.cluster);
+			}
+			if (element instanceof OcmResourceDefinition) {
+				// children of an ocm resource definition are ocm resources
+				return this.getOcmResources(element.resourceSpec);
+			}
+		}
+		// top level children are the connected clusters
+		return this.connectedClusters;
+	}
+
+	private updateK8SConfig(): void {
+		this.kubeConfig.loadFromDefault();
+		this.connectedClusters = this.kubeConfig.clusters.map(cluster => new ConnectedCluster(cluster));
+	}
+
+	private async getOcmResourceDefinitions(cluster: k8s.Cluster): Promise<OcmResourceDefinition[]> {
+		this.kubeConfig.setCurrentContext(cluster.name);
+		let k8sExtApi = this.kubeConfig.makeApiClient(k8s.ApiextensionsV1Api);
+		let apiResponse = await k8sExtApi.listCustomResourceDefinition();
+		if (apiResponse.response.statusCode !== 200) {
+			return Promise.resolve([]);
+		}
+		return Promise.resolve(
+			apiResponse.body.items
+			.filter(item => item.spec.group.includes('open-cluster-management'))
+			.map(item => new OcmResourceDefinition(item.spec)));
+	}
+
+	private async getOcmResources(spec: k8s.V1CustomResourceDefinitionSpec): Promise<OcmResource[]> {
+		let k8sCustomObjApi = this.kubeConfig.makeApiClient(k8s.CustomObjectsApi);
+
+		var listResourcesPromises: Promise<void>[] = [];
+		var customResources: OcmResource[] = [];
+
+		if (spec.scope === 'Namespaced') {
+			await this.getNamespacedResourceLists(spec, k8sCustomObjApi, listResourcesPromises, customResources);
+		} else {
+			await this.getClusterResourceLists(spec, k8sCustomObjApi, listResourcesPromises, customResources);
+		}
+
+		await Promise.allSettled(listResourcesPromises);
+		return Promise.resolve(customResources);
+	}
+
+	private async getNamespacedResourceLists(
+		spec: k8s.V1CustomResourceDefinitionSpec,
+		k8sCustomObjApi: k8s.CustomObjectsApi,
+		listResourcesPromises: Promise<void>[],
+		customResources: OcmResource[]): Promise<void> {
+
+		let k8sCoreApi = this.kubeConfig.makeApiClient(k8s.CoreV1Api);
+		let namespacesApiResponse = await k8sCoreApi.listNamespace();
+		if (namespacesApiResponse.response.statusCode === 200) {
```

Comment:
what about error handling?

Reply:
not for the demo, nither does testing.
for now, for error in k8s, an empty list is returned to not f-up the demo.